Fix content-type params with unquoted specials per RFC 2045, section 5.1

dcormier opened the pull request:
This allowed handling Content-Type headers with parameters that contain unquoted special characters, like this:

	Content-Type: text/calendar; method=text/calendar

In that example, the `method` parameter contains a character that needs to be quoted (`/`), but is not.

This change reformats such parameters so they are quoted properly.

jhillyerd commented on header.go:
```diff
@@ -270,16 +269,162 @@ func fixMangledMediaType(mtype, sep string) string {
 }
 
-// fixUnquotedSpecials as defined in https://www.w3.org/Protocols/rfc1341/4_Content-Type.html
+func consumeParam(s string) (consumed, rest string) {
```
I think a brief comment (with a sample input/output) for this function would help people picture what it does.

dcormier commented on header.go:
```diff
@@ -270,16 +269,162 @@ func fixMangledMediaType(mtype, sep string) string {
+	valueQuoteNeeded := false
+
+findValueStart:
+	for i = range s {
```
In this context `s`  should only contain single-byte characters (I've yet to see _that_ violated; but some email client will prove me wrong). But this change wouldn't hurt anything.

dcormier commented:
Yeah, it was a bigger project than I would've liked, given what the problem was. But this should be pretty resilient.

Code review comments addressed.

jhillyerd commented:
Looks great, thanks

dcormier commented:
When I address code review comments, do you prefer that I squash my commits and force push, or add a new commit so you can more easily spot the diff for addressing the code review?

jhillyerd commented:
Individual commits is fine, it GitHub has added a "what changed since you last looked" button which made it easy to see the nit-fixes.  GH makes it easy to squash during the merge.  But if force-push works better for you to start working on your next change, feel free to use that and we'll see how well GH supports it.
